smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/ProtocolGenerator.kt

```diff
@@ -128,6 +128,11 @@ interface ProtocolGenerator {
      */
     fun generateProtocolClient(ctx: GenerationContext)
 
+    /**
+     * Generate smoke tests defined on the service model
+     */
+    fun generateSmokeTests(ctx: GenerationContext)
```

Comment:
This gets implemented both in smithy-swift and aws-sdk-swift, each generating tests in generic Smithy way vs. AWS way, respectively.